telegram/ext/dispatcher.py

```diff
@@ -229,30 +236,89 @@ def _pooled(self):
                 break
 
             promise.run()
```

Comment:
@Bibo-Joshi,  you wrote quite a nice explanation on one of the threads, but it is now outdated, so can't continue the discussion there.
The exaplanation related to:


1. Why is there a promise.error_handler: when is it used and why?
2. What's the difference between having promise.update and not having one?

However, even with this explanation I'm still a bit confused. I understand that you want to take a different approach for implementation which is fine.
What I do ask you to do, is to choose the implementation you prefer, but before actually implementing it, write a big nice comment on top of the function detailing:
1. What are the various inputs expected
2. How the input going to be handled.

I believe it will help making sense in this complex function.

(and yes, this is the problem with complex, you need to be very precise on what's going on inside it)